### ISSUE TITLE
renameColumn async don't work on my migration flow (due to async i think)

### ISSUE THREAD

**Issue by keep-faith** [sequelize/sequelize#8873]
## What are you doing?

on first migration i rename the column, then i change the type on the following one
migration 1
```js
up: (queryInterface, Sequelize) => {
    return [
      queryInterface.renameColumn('UserReferrals', 'percentage', 'percReward')
    ];
  },
```
migration 2
```js
up: (queryInterface, Sequelize) => {
    return queryInterface.changeColumn('UserReferrals', 'percReward', {
      allowNull: false,
      type: Sequelize.DOUBLE
    });
  },
```

## What do you expect to happen?
rename and change type

## What is actually happening?
i usually have this error when sequelize executes the second migration:

```
Unhandled rejection Error: Table UserReferrals doesn't have the column percReward
```
__Dialect:__ postgres
__Sequelize version:__ 4.28.6

**Comment by keep-faith** [comment]
To me it makes the renameColumn not unusable, but surely untrustable because you never know when you gonna have a situation like this, specially when you're working with a team

```
queryInterface.sequelize.query(`alter table "UserReferrals" rename column "percentage" to "percReward";`),
```

**Comment by sushantdhiman** [comment]
It works properly for me

```js
  await Test.sync({ force: true });

  await sequelize.queryInterface.renameColumn('Tests', 'a', 'percReward');

  await sequelize.queryInterface.changeColumn('Tests', 'percReward', {
    allowNull: false,
    type: Sequelize.DOUBLE
  });
```

```sql
Executing (default): DROP TABLE IF EXISTS "Tests" CASCADE;
Executing (default): CREATE TABLE IF NOT EXISTS "Tests" ("id"   SERIAL , "a" INTEGER, "b" INTEGER, "c" INTEGER, "createdAt" TIMESTAMP WITH TIME ZONE NOT NULL, "updatedAt" TIMESTAMP WITH TIME ZONE NOT NULL, PRIMARY KEY ("id"));
Executing (default): SELECT i.relname AS name, ix.indisprimary AS primary, ix.indisunique AS unique, ix.indkey AS indkey, array_agg(a.attnum) as column_indexes, array_agg(a.attname) AS column_names, pg_get_indexdef(ix.indexrelid) AS definition FROM pg_class t, pg_class i, pg_index ix, pg_attribute a WHERE t.oid = ix.indrelid AND i.oid = ix.indexrelid AND a.attrelid = t.oid AND t.relkind = 'r' and t.relname = 'Tests' GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;
Executing (default): SELECT pk.constraint_type as "Constraint", c.column_name as "Field", c.column_default as "Default", c.is_nullable as "Null", (CASE WHEN c.udt_name = 'hstore' THEN c.udt_name ELSE c.data_type END) || (CASE WHEN c.character_maximum_length IS NOT NULL THEN '(' || c.character_maximum_length || ')' ELSE '' END) as "Type", (SELECT array_agg(e.enumlabel) FROM pg_catalog.pg_type t JOIN pg_catalog.pg_enum e ON t.oid=e.enumtypid WHERE t.typname=c.udt_name) AS "special" FROM information_schema.columns c LEFT JOIN (SELECT tc.table_schema, tc.table_name, cu.column_name, tc.constraint_type FROM information_schema.TABLE_CONSTRAINTS tc JOIN information_schema.KEY_COLUMN_USAGE  cu ON tc.table_schema=cu.table_schema and tc.table_name=cu.table_name and tc.constraint_name=cu.constraint_name and tc.constraint_type='PRIMARY KEY') pk ON pk.table_schema=c.table_schema AND pk.table_name=c.table_name AND pk.column_name=c.column_name WHERE c.table_name = 'Tests' AND c.table_schema = 'public'
Executing (default): ALTER TABLE "Tests" RENAME COLUMN "a" TO "percReward";
Executing (default): ALTER TABLE "Tests" ALTER COLUMN "percReward" SET NOT NULL;ALTER TABLE "Tests" ALTER COLUMN "percReward" DROP DEFAULT;ALTER TABLE "Tests" ALTER COLUMN "percReward" TYPE DOUBLE PRECISION;

```

Why you are returning an array in first migration ?

**Comment by keep-faith** [comment]
i never had this error before, but I have something like 50 migrations and I started a new DB from zero to run my api on production. So I runned db:migrate to execute all migrations.. then i got the error and runned db:migrate:undo:all
After this if you still db:migrate and db:migrate:undo:all it breaks a lot of times.

Then I did the previously mentioned workaround and started db:migrate and db:migrate:undo:all again. The results were good, but still less often breaking.

My thoughts are about async problem but I'm not sure about it.

Ps. I only have the problem with the renameColumn operation
